### PR TITLE
fix(main): fold in-meeting notes into the summary again

### PR DESCRIPTION
**Pull request by luchfilip**
## Problem

Notes typed during a meeting (the "In-app note-taking" feature) never reach the AI summary. The model has no idea the note existed.

## Root cause

The notes sidecar is written and read from **two different directories**:

- **Writer** — `save-meeting-notes` writes to the user-data output dir:
  `path.join(_cachedCustomStoragePath || getUserDataDir(), 'output', '<stem>_notes.txt')`
- **Reader** — the stop-recording handler (`process-system-audio-recording`) looked for it in the **read-only app bundle**:
  `path.join(getBackendCwd(), '_internal', 'output', '<stem>_notes.txt')`

That bundle path doesn't exist for packaged users, so `fs.existsSync(notesFile)` always returned false, `--notes` was never passed to `simple_recorder.py`, `notes_text` stayed `None`, and the note was dropped from the summary.

This looks like a half-completed move: the writer was intentionally moved out of the read-only bundle (see the comment on `save-meeting-notes`), but the reader was left pointing at the old location. The backend safety net that would have caught it — `MeetingPipeline._load_user_notes()` in `simple_recorder.py` — is currently dead code (never called), so nothing masked the mismatch.

Verified on a packaged 0.5.8 install: the note file was sitting in `~/Library/Application Support/stenoai/output/` while the reader looked under `Steno.app/Contents/Resources/stenoai/_internal/output/`, which did not exist.

## Fix

Give the notes path a single source of truth so the writer and reader can't drift apart again:

- **`app/notes-file.js`** (new) — `safeSessionStem()` (kept in sync with the Python `re.sub(r'[^a-zA-Z0-9_-]', '_', name)`) and `userNotesFilePath(outputDir, name)`.
- **`app/main.js`** — new `getOutputDir()` helper; both `save-meeting-notes` and the stop-recording reader now resolve the path via `userNotesFilePath(getOutputDir(), name)`.

## Tests

- **`app/notes-file.test.js`** (new, `node:test`) — stem sanitization (incl. Unicode + null/undefined), sidecar path shape, a regression test asserting writer and reader resolve to the **same** path for the same dir+name, and a guard that the path never lands inside the app bundle (`_internal`).
- Wired into the `test:unit` script.

```
# tests 6
# pass 6
# fail 0
```

## Follow-up (not in this PR)

`_load_user_notes()` in `simple_recorder.py` is defined but never called. Wiring it up as a backend-side fallback would make the pipeline robust even if the Electron path regresses again. Happy to do it in a separate PR if you'd like.

---
## Summary by cubic
Fixes dropped in-meeting notes and flushes both last-second edits and cleared notes on Stop so summaries always reflect user notes. Save errors are surfaced without blocking processing, and filename sanitization now matches Python across Unicode.

- **Bug Fixes**
  - Add `app/notes-file.js` with `safeSessionStem()` (uses the `u` flag for astral-char parity) and `userNotesFilePath()`; single source for notes paths.
  - Add `getOutputDir()` in `app/main.js` and route both `save-meeting-notes` and stop-recording through `userNotesFilePath(getOutputDir(), name)` instead of the read-only bundle path.
  - Flush notes before processing to close the debounce race; write '' to overwrite stale sidecars; unwrap save-note Result failures to surface errors, but still process.
  - Tests: add `app/notes-file.test.js` and `app/renderer/src/lib/notesFlush.test.ts` (ordering, undefined skip, empty overwrite, best-effort on error); wire `notes-file.test.js` into `test:unit`.

<sup>Written for commit 197ea04a6e2d12e957b7855accf7424787ab0c92. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/ruzin/stenoai/pull/330?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by luchfilip** [comment]
Notes were not at all considered for the summary thus the PR. Full credits to Claude Opus 4.8. Tested the notes again after the local fix: LLM correctly accounted for the written notes + transcript.

File: app/main.js [hunk 1]
```diff
@@ -7060,9 +7067,12 @@ ipcMain.handle('process-system-audio-recording', async (event, audioFilePath, se
...
 
-    // Check for user notes file
-    const safeName = actualSessionName.replace(/[^a-zA-Z0-9_-]/g, '_');
-    const notesFile = path.join(getBackendCwd(), '_internal', 'output', `${safeName}_notes.txt`);
+    // Check for user notes file. Resolve it from the SAME output dir that
+    // 'save-meeting-notes' wrote it to (getOutputDir / userNotesFilePath) — not
+    // the read-only bundle dir. The old getBackendCwd()/_internal/output path
+    // never existed for packaged users, so the lookup always missed, --notes was
+    // never passed, and in-meeting notes were silently dropped from the summary.
+    const notesFile = userNotesFilePath(getOutputDir(), actualSessionName);
```
[the remark this answers is not in the document]
Comment (ruzin) [reply app/main.js:7069]: @luchfilip could you resolve these comments as well, just a quick pass with opus should do it

[the remark this answers is not in the document]
Comment (luchfilip) [reply app/main.js:7069]: on it

[the remark this answers is not in the document]
Comment (luchfilip) [reply app/main.js:7069]: this was valid concern. a note typed in the last 500 ms could have been lost. We should wait for completion first.

[the remark this answers is not in the document]
Comment (luchfilip) [reply app/main.js:7069]: Fixed the other valid concerns with graceful fallbacks, edge cases handling and new tests.

**Review by Optic00 — COMMENTED** [review]
## Review

The root-cause analysis in the PR body checks out against git history, and the fix direction is right. I verified the story end to end:

- #87 introduced writer AND reader at the bundle path `_internal/output` (worked in dev, writer failed silently in packaged apps).
- c0eb5d7 (PR #201) moved only the writer to the user-data output dir - the half-completed move this PR describes.
- #226 (retiring the Python mic record path) deleted the only call sites of `_load_user_notes`, which is why the backend safety net never masked the drift.

Also verified the new wiring: `bridge.meetings.saveNotes` -> `save-meeting-notes`, `getLiveDraft` semantics (`ensure()` creates drafts with `notes: ''`, so the `''`-flush correctly blanks stale sidecars in the normal flow), `unwrap` Result handling, and that Python's `--notes` path strips empty files to `None`, so an empty sidecar is harmless. The `u`-flag parity with Python's `re.sub` is correct. Tests are the right shape and wired into `test:unit`.

Two things I'd fix before merge (both cheap):

### 1. `delete-meeting` still computes the sidecar path inline (third site, old sanitizer)

In the `delete-meeting` handler (app/main.js, the `absolutePaths` block):

```js
const safeName = sessionName.replace(/[^a-zA-Z0-9_-]/g, '_');
absolutePaths.push(path.join(outputDir, `${safeName}_notes.txt`));
```

This is exactly the drift class the new helper is meant to eliminate, and it drifts already: no `u` flag, so for a session named `a😀b` the writer creates `a_b_notes.txt` while delete-meeting tries to remove `a__b_notes.txt`. Deleted meetings then leave orphaned sidecars behind, which feeds the stale-sidecar problem below. Suggest `userNotesFilePath(outputDir, sessionName)` here too (the helper takes the dir as a param, so the summary-file-derived outputDir keeps working).

### 2. The stale-sidecar failure mode is now live - add a main-side guard

Before this PR the reader never found any sidecar, so leftover `*_notes.txt` files were inert. Now they're read. Nothing deletes a sidecar after processing (only meeting deletion does), and session names collide readily (the default is `Note`). The `''`-flush covers the common path, but it's renderer-best-effort: if the draft is absent (`getDraftNotes` returns `undefined`, so the flush is skipped) or the flush write fails (`onFlushError` logs and processing continues), the stop handler happily folds a previous same-name session's notes into the new summary. Wrong-meeting notes in a summary is a worse failure than missing notes.

One main-side line kills the entire class regardless of renderer behavior: delete (or truncate) the sidecar for the session name at recording start (`start-recording-ui`), so a sidecar can only ever contain notes written during the current session. An mtime check in the reader (only pass `--notes` if the file is newer than recording start) would work too, but delete-on-start is simpler and also covers crash/rescue paths.

### Minor, non-blocking

- Narrow race: stop recording A, immediately start "+ New note" with the same default name. The start flow clears the draft and `ensure()` recreates it empty, so A's close callback can flush `''` over A's debounced notes before main reads the sidecar. The delete-on-start guard from point 2 doesn't fix this one (it's the same-name-overlap case), but it's a tight window and strictly no worse than main today, where notes are always dropped.
- main doesn't trim the session name (`sessionName || 'Note'`) while the renderer's optimistic path does (`name.trim() || 'Note'`). Pre-existing, and writer/reader converge on the same raw name after queue reconciliation, so it doesn't break this fix - just noting it since you were in the area.
- The follow-up you propose (rewiring `_load_user_notes` as a backend fallback) sounds right, and with the shared stem helper on the JS side plus the same regex in Python they'd stay in sync.

Nice work tracking this down - the verified-on-packaged-0.5.8 repro in the description is exactly what made this reviewable.

**Review by ruzin — APPROVED** [review]
Reviewed the full diff (medium-effort correctness + cleanup pass). Fixes a genuine data-loss bug: writer/reader sidecar path drift silently dropped in-meeting notes from summaries. The single-source-of-truth helper (userNotesFilePath) plus the flush-before-process ordering are correct; regex parity with Python is real and tested; empty-name mismatch is unreachable (coerced to 'Note' upstream). All T2/pipeline lanes green. No actionable findings.

**Comment by ruzin** [comment]
@luchfilip your change is going in today's release, thanks for the contribution

**Comment by ruzin** [comment]
@luchfilip join the discord if you https://discord.com/invite/DZ6vcQnxxu haven't already
